File: ABC/350/a.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string s;
    cin >> s;

    string t = s.substr(3, 3);
    int num = stoi(t);

    if (num >= 1 && num < 350 && num != 316) cout << "Yes" << endl;
    else cout << "No" << endl;

    return 0;
}

File: ABC/350/a.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        string S = Console.ReadLine();

        int num = int.Parse(S.Substring(3,3));

        if (num >= 1 && num <= 349 && num != 316)
        {
            Console.WriteLine("Yes");
        }
        else
        {
            Console.WriteLine("No");
        }
    }
}
